includes/class-wc-stripe-domain-registration.php

```diff
@@ -260,15 +265,15 @@ private function make_domain_registration_request( $secret_key ) {
 		if ( 200 !== $response['response']['code'] ) {
 			$parsed_response = json_decode( $response['body'] );
 
-			$this->apple_pay_verify_notice = $parsed_response->error->message;
+			$this->apple_pay_verify_notice = $parsed_response->apple_pay->status_details->error_message ?? '';
```

Comment:
I confirmed that this is the way to fetch an error message, however, the response code in this case was a 200.

<p align="center">
<img width="467" alt="Screenshot 2024-09-11 at 1 50 27 PM" src="https://github.com/user-attachments/assets/222904ca-12d7-4584-8bdb-754b24437af7">
</br>
<sup><strong><code>POST</code></strong> <code>/v1/payment_method_domains</code> response when the file is missing.</sup>

To get that response, I manually tweaked the domain to mock a failure. 

Given the response is a `200` code, this `$this->apple_pay_verify_notice` is never set. 

I don't know what the intent is with this though because as far as I can tell, the `$this->apple_pay_verify_notice` variable is only used to display an admin notice and it's never stored anywhere so it would only be shown on the admin request that registered the domain. 

Anyway, I'm not sure if you had a different method of testing this, but I just wanted to flag it incase we intend on changing this. 
</p>

Reply:
Good catch, James! I could not find a simple way to test the exception. Thanks for doing it! I have updated the code to check the content of the error message instead of the status code. This is similar to what was done [here](https://github.com/Automattic/woocommerce-payments/pull/9139/files).